TOOLS/ce/src/main/scala/org.opalj.ce/Comment.scala

```diff
@@ -0,0 +1,144 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj.ce
+
+import scala.collection.mutable.ListBuffer
+
+/**
+ * Container for the comments of a config node.
+ */
+class Comment {
+    val commentBuffer: ListBuffer[String] = ListBuffer[String]()
+    var constraints: ListBuffer[String] = ListBuffer[String]()
```

Comment:
You are right. During the replacement of the subclasses, the entire constraints field got rreplaced. That explains the var. However the building process from multiple lines require it to be a mutable list.
I have restructured the "comment", so that the replacements of the both now include the source class as well and do use the modifying options of the mutable.ListBuffer